lib/index.js

```diff
@@ -273,6 +273,48 @@ exports.createStorage = async (ctx) => {
     }
 };
 
+exports.deleteStorage = async (ctx) => {
+    debug("DELETE STORAGE");
+
+    // Find the Client
+    const issuerId = `uri:fusebit-template:${ctx.functionId}:${ctx.body.subscriptionId}:${ctx.body.boundaryId}:${ctx.body.functionId}`;
+    const subject = 'client-1';
+    debug('Looking up Client ID', issuerId);
+    const response = await Superagent.get(`${ctx.body.baseUrl}/v1/account/${ctx.body.accountId}/client?issuerId=${
+            encodeURIComponent(issuerId)
+        }&subject=${subject}&include=all`)
+        .set("Authorization", ctx.headers['authorization']); // pass-through authorization
+    const client = response.body.items && response.body.items[0];
+    debug('Found client', client);
+
+    if (client) {
+        // Delete the storage
+        const allow = (client.access && client.access.allow || []);
+        for (var i = 0; i < allow.length; i++) {
+            if (allow[i].action.indexOf('storage:') === 0) {
+                await Superagent.delete(`${ctx.body.baseUrl}/v1${allow[i].resource}`)
```

Comment:
Can we pass in the storage key to delete explicitly instead? This heuristic assumes no one will ever modify the permissions of the client we created outside of the logic we own. We may inadvertently delete storage we don't "own".

Reply:
I'm not sure this is part of this change, but I would like to retain the option for multiple pieces of storage in a given function, with possibly different permissions. Something to think about.

Reply:
@tjanczuk so to get this we'll need to get the function metadata. If we do so then we will also have access to all of the following, so we should probably switch over delete to use them instead of the current heuristic? 

```
fusebit_storage_issuer_id
fusebit_storage_subject
fusebit_storage_id
fusebit_storage_audience
fusebit_storage_account_id
fusebit_storage_subscription_id
```

Reply:
Or maybe, you can delete storage from within the *uninstall* logic in the add-on handler since you are calling it already from /uninstall on the addon template.

Reply:
@tjanczuk either way though, do you agree to switching it to use the above variables instead?

Reply:
I think it makes sense

Reply:
@tjanczuk Can you look at the latest and let me know if it looks good overall? I'm about to go test this to make sure the addons still work, and I want to make sure we're on the same page before I spend the time.